### PR TITLE
Verify production/staging server TLS when using poller serve

### LINKED ISSUE #39: Add monitoring agent TLS cert
- [ ] Agent TLS Cert
- [ ] Add --insecure option to skip TLS certificate checking, but log when this option is used

### PR DESCRIPTION
**Pull request by itzg**
- [x] Add unit tests

Resolves #39

### CODE REVIEW

File: commands/serve.go [hunk 1]
```diff
@@ -48,18 +52,40 @@ func HandleInterrupts() chan os.Signal {
...
 }
 
+func LoadRootCAs(insecure bool) *x509.CertPool {
+	if insecure {
+		log.Warn("Insecure TLS connectivity is enabled. Make sure you TRUST the farend host")
+		return nil
+	} else {
+		devCAPath := os.Getenv(config.EnvDevCA)
+		isStaging := os.Getenv(config.EnvStaging)
+		if isStaging == "1" {
```
Comment (dimtruck) [inline commands/serve.go]: Magic number.  Do we want to set it to a const somewhere?  Seems to be only used once so probably not a big deal.

File: commands/serve_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// limitations under the License.
+//
+
+
+package commands_test
+
+import (
+	"testing"
+	"github.com/racker/rackspace-monitoring-poller/commands"
+	"github.com/stretchr/testify/assert"
```
Comment (dimtruck) [inline commands/serve_test.go]: :)

File: config/certs.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+import (
+	"crypto/x509"
+	"github.com/Sirupsen/logrus"
+	"io/ioutil"
+)
+
+// /C=US/ST=Texas/L=San Antonio/O=Rackspace US, Inc/OU=Cloud Monitoring/CN=monitoring-ca.rackspace.com
+// /C=US/ST=Texas/L=San Antonio/O=Rackspace US, Inc/OU=Cloud Monitoring/CN=monitoring-test-ca.rackspace.com
+var productionCAPems = `
```
Comment (dimtruck) [inline config/certs.go]: Newbie question - but shouldn't these be secret?

File: contrib/local-endpoint.cfg [hunk 4]
```diff
@@ -1,3 +1,3 @@
 monitoring_token 0000000000000000000000000000000000000000000000000000000000000000.7777
 monitoring_id agentA
-monitoring_endpoints endpoint:55000
+monitoring_endpoints localhost:55000
```
Comment (dimtruck) [inline contrib/local-endpoint.cfg]: This'll break if using docker-compose :(

Comment (dimtruck) [inline contrib/local-endpoint.cfg]: Although, maybe not.  Let me test again.  It's quite possible I just changed it to match the container name, which may not have been necessary

File: commands/serve.go (hunk 1 above)
Comment (itzg) [inline commands/serve.go]: Oops, missed that one.

File: config/certs.go (hunk 3 above)
Comment (itzg) [inline config/certs.go]: These are only the public certificates, so no worries with source controlling this. Now, the keys are ultra-sensitive...which is why the README requires generating your own.

File: contrib/local-endpoint.cfg (hunk 4 above)
Comment (itzg) [inline contrib/local-endpoint.cfg]: Oh...that's why it was `endpoint`. Let's create a separate cfg file for docker-compose

Comment (dimtruck) [inline contrib/local-endpoint.cfg]: Yeah, it broke with localhost... cause it's in a different container.  +1 on a different cfg file.

Comment (itzg) [inline contrib/local-endpoint.cfg]: Working on it and will tweak the `docker-compose.yml` accordingly...

**Comment by itzg** [comment]
@dimtruck ready for re-review, esp of the docker bits

**Comment by rphillips** [comment]
nice job... only comment would be to see if the ConnectionStream should own the TLSConfig object.

**Comment by itzg** [comment]
@rphillips  I remember now...the tlsConfig is technically unique per connection since the farend `addr` needs to be specified/verified. Instead of Connection owning keeping around the instance, I could just pass it into the [Connection.Connect call](https://github.com/racker/rackspace-monitoring-poller/blob/fd9b99c281c88f1d875ce6c9c7044d2e321d54eb/poller/connection.go#L69-L69).

**Comment by rphillips** [comment]
ah hah!. ok

**Comment by rphillips** [comment]
+1

**Comment by rphillips** [comment]
+1

**Pull request merged**